coap/api/server.go

```diff
@@ -0,0 +1,126 @@
+package api
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	mux "github.com/dereulenspiegel/coap-mux"
+	"github.com/mainflux/mainflux/coap"
+	manager "github.com/mainflux/mainflux/manager/client"
+
+	gocoap "github.com/dustin/go-coap"
+)
+
+func authKey(opt interface{}) (string, error) {
+	if opt == nil {
```

Comment:
You can remove this check. It is also performed as part of the type assertion below.